iam.tf

```diff
@@ -108,7 +108,7 @@ data "aws_iam_policy_document" "assume_role" {
 resource "aws_iam_role" "logs" {
   count = var.logging_enabled && var.create_logs_role ? 1 : 0
 
-  name                 = coalesce(var.logs_role_name, "${var.name}-logs")
+  name                 = "${coalesce(var.logs_role_name, "${var.name}-logs")}${var.role_suffix}"
```

Comment:
You can parametrize `var.logs_role_name` to be unique to create non-colliding resources.